src/Confluent.Kafka.Avro/AvroSerializer.cs

```diff
@@ -40,118 +36,25 @@ namespace Confluent.Kafka.Serialization
     /// </remarks>
     public class AvroSerializer<T> : ISerializer<T>
     {
-        private SpecificWriter<T> avroWriter;
-
-        private HashSet<string> topicsRegistered = new HashSet<string>();
+        private bool autoRegisterSchema = true;
+        private int initialBufferSize = DefaultInitialBufferSize;
+        private bool isKeySerializer;
+        private ISchemaRegistryClient schemaRegistryClient;
 
         private bool disposeClientOnDispose;
 
-        private string writerSchemaString { get; set; }
+        private IAvroSerializerImpl<T> serializerImpl;
 
         private const string InitialBufferSizePropertyName = "avro.serializer.buffer.bytes";
 
         private const string AutoRegisterSchemaPropertyName = "avro.serializer.auto.register.schemas";
 
-        /// <summary>
-        ///     The SchemaId corresponding to type <see cref="T"/>, or null if 
-        ///     the serializer has yet to be invoked.
-        /// </summary>
-        public int? SchemaId { get; private set; } = null;
-
-        /// <summary>
-        ///	    The <see cref="ISchemaRegistryClient"/> instance used for communication
-        ///	    with Confluent Schema Registry.
-        /// </summary>
-        public ISchemaRegistryClient SchemaRegistryClient { get; private set; }
-
         /// <summary>
         ///     The default initial size (in bytes) of buffers used for message 
         ///     serialization.
         /// </summary>
         public const int DefaultInitialBufferSize = 128;
 
-        /// <summary>
-        ///     True if the serializer will attempt to auto-register un-recognized schemas
-        ///     with Confluent Schema Registry, false if not.
-        /// </summary>
-        public bool AutoRegisterSchema { get; private set; } = true;
```

Comment:
i also removed these. they're actually still applicable in all cases, but i feel there's no real need to expose these and safer not to.